brush-core/src/shell.rs

```diff
@@ -178,6 +178,18 @@ pub struct CreateOptions {
     pub verbose: bool,
     /// Maximum function call depth.
     pub max_function_call_depth: Option<usize>,
+    /// The parser implementation to use.
+    pub parser: ParserImplementation,
+}
+
+/// Represents the parser implementation.
+#[derive(Debug, Clone, Default, clap::ValueEnum)]
+pub enum ParserImplementation {
```

Comment:
Since you're not directly exposing this type through `brush-shell` to clap, does this need to derive `clap::ValueEnum`?

Reply:
No. I wasn't sure about keeping the two enums since they are pretty much identical - what would be better?

Reply:
I think it's debatable, but you can probably just keep the "inner" one in `brush-core` and directly expose it via `brush-shell` without the wrapper.